fang/src/asynk/async_queue.rs

```diff
@@ -165,165 +171,49 @@ where
-}
+use tokio::sync::Mutex;
 
 #[cfg(test)]
-#[async_trait]
-impl AsyncQueueable for AsyncQueueTest<'_> {
-    async fn find_task_by_id(&mut self, id: Uuid) -> Result<Task, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::find_task_by_id_query(transaction, id).await
-    }
-
-    async fn fetch_and_touch_task(
-        &mut self,
-        task_type: Option<String>,
-    ) -> Result<Option<Task>, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::fetch_and_touch_task_query(transaction, task_type).await
-    }
-
-    async fn insert_task(&mut self, task: &dyn AsyncRunnable) -> Result<Task, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        let metadata = serde_json::to_value(task)?;
-
-        let task: Task = if !task.uniq() {
-            AsyncQueue::<NoTls>::insert_task_query(
-                transaction,
-                metadata,
-                &task.task_type(),
-                Utc::now(),
-            )
-            .await?
-        } else {
-            AsyncQueue::<NoTls>::insert_task_if_not_exist_query(
-                transaction,
-                metadata,
-                &task.task_type(),
-                Utc::now(),
-            )
-            .await?
-        };
-        Ok(task)
-    }
+static ASYNC_QUEUE_DB_TEST_COUNTER: Mutex<u32> = Mutex::const_new(0);
 
-    async fn schedule_task(&mut self, task: &dyn AsyncRunnable) -> Result<Task, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        let metadata = serde_json::to_value(task)?;
-
-        let scheduled_at = match task.cron() {
-            Some(scheduled) => match scheduled {
-                CronPattern(cron_pattern) => {
-                    let schedule = Schedule::from_str(&cron_pattern)?;
-                    let mut iterator = schedule.upcoming(Utc);
-
-                    iterator
-                        .next()
-                        .ok_or(AsyncQueueError::CronError(CronError::NoTimestampsError))?
-                }
-                ScheduleOnce(datetime) => datetime,
-            },
-            None => {
-                return Err(AsyncQueueError::CronError(
-                    CronError::TaskNotSchedulableError,
-                ));
+#[cfg(test)]
+impl AsyncQueue<NoTls> {
+    /// Provides an AsyncQueue connected to its own DB
+    pub async fn test() -> Self {
+        const BASE_URI: &str = "postgres://postgres:postgres@localhost";
+        let mut res = Self::builder()
+            .max_pool_size(1_u32)
+            .uri(format!("{}/fang", BASE_URI))
+            .build();
+
+        let mut new_number = ASYNC_QUEUE_DB_TEST_COUNTER.lock().await;
+        res.connect(NoTls).await.unwrap();
+
+        let db_name = format!("async_queue_test_{}", *new_number);
+        *new_number += 1;
+
+        let create_query = format!("CREATE DATABASE {} WITH TEMPLATE fang;", db_name);
```

Comment:
can we add some logs like.

"Creating database"
"Dropping database"

etc

Reply:
sure

Reply:
[merge added some logs](https://github.com/ayrat555/fang/pull/133/commits/7c4c442d8f09d6a7680cc04caaf757e52e24b45e)